.eslintrc

```diff
@@ -2,37 +2,53 @@
     "root": true,
     "parser": "@typescript-eslint/parser",
     "parserOptions": {
-      "ecmaVersion": 2018,
-      "sourceType": "module",
-      "ecmaFeatures": {
-        "impliedStrict": true,
-        "jsx": true
-      },
-      "project": false
+        "ecmaVersion": 2018,
+        "sourceType": "module",
+        "ecmaFeatures": {
+            "impliedStrict": true,
+            "jsx": true
+        },
+        "project": "tsconfig.json"
     },
     "plugins": [
-      "@typescript-eslint",
-      "react",
-      "react-hooks"
+        "@typescript-eslint",
+        "react",
+        "react-hooks"
     ],
     "extends": [
-      "eslint:recommended",
-      "plugin:@typescript-eslint/recommended",
-      "plugin:@typescript-eslint/strict",
-      "plugin:@typescript-eslint/stylistic",
-      "plugin:react/recommended",
-      "plugin:react/jsx-runtime",
-      "plugin:prettier/recommended",
-      "plugin:react-hooks/recommended"
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-type-checked",
+        "plugin:@typescript-eslint/strict",
+        "plugin:@typescript-eslint/stylistic",
+        "plugin:react/recommended",
+        "plugin:react/jsx-runtime",
+        "plugin:prettier/recommended",
+        "plugin:react-hooks/recommended"
     ],
     "rules": {
-      "no-console": 1,       // Means warning
-      "prettier/prettier": 2,
-
-      "@typescript-eslint/no-inferrable-types": "off",
-
-      // Previously warn but now errors
-      "@typescript-eslint/no-non-null-assertion": "warn",
-      "@typescript-eslint/no-explicit-any": "warn"
+        "no-console": 1, // Means warning
+        "prettier/prettier": 2,
+        "@typescript-eslint/no-inferrable-types": "off",
+        // Previously warn but now errors
+        "@typescript-eslint/no-non-null-assertion": "warn",
+        "@typescript-eslint/no-explicit-any": "warn",
+        // To reenable as error progressively
```

Comment:
To avoid adding errors everywhere I put those as warn, but those are relevant errors that we want to fix